modules/apps/object/object-api/src/main/java/com/liferay/object/exception/RequiredObjectFieldRelationshipException.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.exception;
+
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Larissa Ribeiro
+ */
+public class RequiredObjectFieldRelationshipException extends PortalException {
+
+	public RequiredObjectFieldRelationshipException(
+		ObjectDefinition objectDefinition) {
+
+		super(
+			StringBundler.concat(
+				"Deletion not allowed because a relationship field is being ",
+				"used in the object definition ",
+				objectDefinition.getShortName()));
+
+		_arguments = Arrays.asList(
+			objectDefinition.getShortName(), objectDefinition.getShortName());
```

Comment:
@larissacribeiro we only need to pass 1 argument here. You can change the language key to: 

```
Deletion not allowed because a relationship field is being used in the object definition "{0}". 
To proceed, introduce a new custom field in "{0}".
```

Reply:
done.